### ISSUE TITLE
Remove pill button in MultiSelect is not accessible

### ISSUE THREAD

**Issue by ernon66** [Sage/carbon#7514]
### Description

The remove pill button in a MultiSelect just reads "remove pill" and it isn't clear which option it refers to which fails WCAG 1.3.1 Info and Relationships

The HTML structure or the button's accessible name should make it clear which pill it removes

### Reproduction

https://carbon.sage.com/?path=/docs/select-multiselect--docs

### Steps to reproduce

Select multiple options in a MultiSelect

With a screen reader, if you go through buttons it'll just read "remove pill" several times

### JIRA ticket numbers (Sage only)

SBS-143005

### Suggested solution

This is a button with just an icon so aria could be acceptable

Allow users to provide aria properties for each button (through Option ?)

Or add an aria-label `${removePillTranslation} ${pillName}` 

Or add an aria-label `${removePillTranslation}` and aria-labelledby `${selfId} ${pillNameId}` 

Maybe there's a better solution with pure HTML

### Carbon version

156.3.1

### React version

v17

### Relevant browsers

Chrome, Safari

### Relevant OSs

MacOS

### Confidentiality

- [x] I confirm there is no confidential or commercially sensitive information included.
